### ISSUE TITLE
Analysis V3A: benchmark music foundation representations for similarity, retrieval, and downstream MIR

### ISSUE THREAD

**Issue by gr-rr** [gr-rr/hello-ai#332]
Parent: #327

## Decision to make
Choose whether hello-ai should add a music-foundation representation layer, and if so which model(s) are practical enough to become the canonical embedding/evidence path.

This is an **evaluation-only execution task**. The candidate list, product questions, and decision criteria are already fixed. Do not perform another open-ended model survey and do not productionize anything in this issue.

## Fixed first-round candidates / checkpoints
Evaluate exactly these unless one is mechanically unavailable:

1. **MERT** — official `m-a-p/MERT-v1-95M` checkpoint. Use 95M first; do not add 330M unless 95M reveals a concrete capacity gap.
2. **MuQ** — official `OpenMuQ/MuQ-large-msd-iter`. Released weights are non-commercial (CC-BY-NC); classify accordingly rather than rejecting the research comparison.
3. **MusicFM** — official paper/FMA checkpoint (`pretrained_fma.pt` + matching stats) as primary comparison. The MSD checkpoint may be noted but is not required unless its terms and operational value are clear.
4. **CLaMP3** — official pretrained configuration(s). Use the documented SAAS/audio and C2/symbolic paths as intended by upstream; do not hack incompatible checkpoints together. Record whether the upstream checkpoint split complicates direct audio↔symbolic comparison.
5. **LAION CLAP** — music-specific checkpoint `music_audioset_epoch_15_esc_90.14.pt` as the generic audio-text comparison baseline, if the official checkpoint remains mechanically loadable.

Use official upstream code/model IDs. Record exact repo commit, package/model version, and file/checkpoint checksum actually used.

## Evaluation harness
Prefer **MARBLE v2** for encoders/tasks it already supports. Do not reimplement MARBLE. If a candidate/product-specific test is not supported cleanly, write a thin local adapter around the official inference API.

Do not modify production engine registries, `capabilities.json`, DB schema, frontend, or deployment config.

Suggested research layout:

```text
backend/evaluation/analysis_v3/foundation/
  README.md
  run.py
  adapters/
    mert.py
    muq.py
    musicfm.py
    clamp3.py
    clap.py
  manifests/
  results/
    mert.json
    muq.json
    musicfm.json
    clamp3.json
    clap.json
  REPORT.md
```

Do not commit downloaded checkpoints, copyrighted audio, virtualenvs, or large caches.

### 1. Segment-to-segment similarity within one work
Use fixed 5–15 second windows. Given one segment, retrieve other windows from the same track. Record nearest-neighbor rankings and qualitative failure cases.

### 2. Cross-work similarity
Use a deterministic rights-safe diversity probe, not only piano. Prefer official Creative-Commons/open research audio such as a small fixed subset of FMA or MTG-Jamendo. Select IDs deterministically and commit only the manifest/metadata, not the audio.

Include materially different musical organizations/instrumentations. These are probes, not app genres.

### 3. Text-to-passage retrieval where supported
Use the same neutral factual prompt set for every text-capable candidate, for example:
- `solo piano`
- `prominent drums and bass`
- `sparse vocal passage`
- `dense distorted guitars`
- `acoustic ensemble`

Do not use subjective prompts as the primary score.

### 4. Cross-representation consistency
CLaMP3 gets an explicit hello-ai test using aligned artifacts from the same Work where available:

```text
audio segment ↔ corresponding MIDI ↔ corresponding MusicXML/score
```

Compare matched-pair ranks against mismatched passages/works. If official CLaMP3 packaging prevents a valid direct comparison, report the limitation rather than inventing a bridge.

### 5. Reference downstream MIR quality
Reuse MARBLE/reference results or a small existing probe where practical. The goal is to contextualize the representation, not to retrain a full suite of heads in this issue.

## Operational measurements
For every candidate record:
- clean install success/failure
- Python requirements
- CPU support
- GPU requirement/benefit
- ARM feasibility if testable or clearly documented
- checkpoint/model size
- additional install size
- runtime downloads
- latency on 10s, 30s, and whole-track input
- peak RAM/VRAM
- output dimensionality and temporal granularity
- deterministic/stable inference behavior
- failure rate

Run CPU measurements even for GPU-oriented models so we know whether current Oracle deployment is plausible. Do not change production topology to accommodate a research candidate.

## Licensing
Record separately:
- code license
- model-weight/checkpoint license
- training/data restrictions explicitly stated upstream

Do not infer weight rights from code license.

## Metrics / outputs
For retrieval tests record at minimum:
- Recall@1 / Recall@5 where paired ground truth exists
- MRR where appropriate
- matched-pair rank for cross-modal tests
- per-query result table / failure examples

For unlabelled within-work similarity, call it a qualitative/product probe, not benchmark accuracy.

## Required report table
`REPORT.md` must contain one row per candidate:

```text
candidate | exact model | license | size | CPU latency | RAM | segment support | audio-text | symbolic | benchmark/reference | product-probe result | decision
```

Every candidate receives exactly one decision:
- `ADOPT`
- `RESEARCH`
- `REJECT`
- `REVISIT`

Then provide **one canonical embedding architecture recommendation**, even if that recommendation is “no production embedding layer yet.”

## Product contract to propose, not implement
Propose an `EmbeddingEvidence` contract covering:
- model + version/checksum
- modality
- artifact/version lineage
- optional time span
- dimensionality
- normalization
- external vector/artifact reference
- provenance

Do not create tables/vector indexes in this issue. #336 owns schema/ERD decisions after concrete research requirements exist.

## Stop conditions
Stop and report rather than widening scope if:
- a candidate cannot be installed after one reasonable upstream-compatible attempt,
- weights are unavailable/ambiguous,
- the model requires hardware unavailable for meaningful measurement,
- benchmark data cannot be lawfully obtained.

Classify the candidate `REVISIT` or `REJECT` with exact evidence. Do not spend hours repairing abandoned research stacks.

## Deliverable
1. reproducible evaluation-only code + manifests
2. machine-readable candidate results
3. `REPORT.md`
4. update #327 with the decision table
5. update `docs/RESEARCH_LANDSCAPE.md` / `docs/ANALYSIS_V3_IMPLEMENTATION_PLAN.md` only with measured findings

No frontend/product integration in this issue.
